store/iavl/store.go

```diff
@@ -275,31 +277,36 @@ func (st *Store) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 			break
 		}
 
+		_, res.Value = tree.GetVersioned(key, res.Height)
 		if req.Prove {
```

Comment:
I'd love to see this being implemented as follows:

```suggestion
		if !req.Prove {
			break
		}
```

This would reduce the indentation level and make the code more readable